maths/segmented_sieve.py

```diff
@@ -4,7 +4,22 @@
 
 
 def sieve(n: int) -> list[int]:
-    """Segmented Sieve."""
+    """
+    Segmented Sieve.
+
+    Examples:

```

Comment:
Please add a doctest where the input itself is a prime number. For example,
```py
>>> sieve(29)
[2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
```
This is to clarify whether the algorithm includes the input number itself or not.

Reply:
In addition, please make sure that the function can correctly handle negative and zero inputs

Reply:
@tianyizheng02 corrected.